doc/udev-rules.org

```diff
@@ -0,0 +1,244 @@
+     18 src/lib/network/clients/save_network.rb
+     19 src/lib/network/edit_nic_name.rb
+     26 src/include/network/lan/udev.rb
+     28 src/lib/network/network_autoyast.rb
+    127 src/modules/LanItems.rb
+#+END_SRC
+
+** Areas NOT in scope
+
+These are related to Udev in the current code but let's not touch them
+in this 1st stage.
+
+*** s390 rules (AY: networking/s390-devices)
+
+are similar but not really
+- set up a virtual device
+- do not include a name(!?)
+*** driver rules (selecting the driver seems broken?)
+the other part of ag_udev_persistent
+not part of AY?
+
+** API
+*** Naming
+
+Anything named "udev" should be code-clean.
+
+LanItems - keep it for now but remember that "item" is too generic
+  yet it combines info about a "nic" and its (stored) config.
+
+"Udev" is an implementation detail, the upper layer should say
+"NameRule"
+
+"Udev name" -> "device name", or "persistent device name"
+
+In LanItems, keep the general interface of item["udev"]["net"]
+so that existing code works, BUT
+  instead of ["net"] which is_a Array<String>
+  make       ["name_rule"] which is_a NameRule
+
+*** copying and identity
+#Items are deeply copied values? NO. GetLanItem and getCurrentItem return the
+original data.
+
+*** LanItems item API
+
+This is strictly speaking above our area of focus, but the current naming is
+so awful that at least a glossary is needed to be able to understand what's
+going on.
+
+**** current -> Integer
+the *index* of the current item
+**** GetLanItem(item_id) => item(item_id) # surprisingly no collisions with "item" lvar
+**** getCurrentItem      => current_item
+**** item_name_rule(item_id) = item(item_id)["udev"]["name_rule"]
+also considering that #item would be an adaptor object
+that would translate #name_rule to ["udev"]["name_rule"]
+Does it need to exist? Nil? NullRule?
+**** current_item_name_rule  =  current_item["udev"]["name_rule"]
+
+*** target API
+This is a sketch of the new API as emerging from the Usage section below
+**** NameRule
+***** @udev [UdevRule]
+***** #matcher= and #matcher(:bus_id or :mac)
+***** #value (case sensitive??)
+***** #name
+udev["NAME"]
+**** NameRules
+***** #pathname
+"/etc/udev/rules.d/70-persistent-net.rules"
+
+*** Usage
+Here I list all the mentions of "udev" in the code
+and sketch out how to write them better.
+**** InstallInfConvertor
+***** AllowUdevModify
+checks if cmdline contains "biosdevname=..."
+**** NetworkLanComplexInclude src/include/network/lan/complex.rb
+***** calls LanItems.update_item_udev_rule!(:bus_id)
+**** Lan#Export
+calls LanItems#Export
+**** lan_auto
+***** ToAY converts the net-udev piece from a hash to an array
+**** NetworkLanHardwareInclude
+it's the Hardware tab
+      device_name = LanItems.current_udev_name
+let's keep that
+**** save_network
+#copy_udev_rules
+
+s390 51* leave that
+
+the rule file needs to be copied from inst-sys to target:
+  need its fs path
+  NameRules#pathname (and use std ruby dirname+basename)
```

Comment:
really? Is it really needed? Maybe we can think one level high and it can be NameRules.load and then save with optional prefix? so API can look like with notes from above
```
name_rules = UdevSerializer.load
UdevSerializer.save(name_rules, prefix: "/mnt")
```

Does it make sense? Or maybe there is another requirements that is still hidden?